### ISSUE TITLE
Quest Data Does Not Save

### ISSUE THREAD

**Issue by rtschaefer** [ironmonk88/monks-enhanced-journal#136]
Every time I was creating a Quest, no matter now often I selected Save (either method, including inside the HTML formatting box), all the text data would simply disappear. I continued to test this by entering in one word in the GM notes tab of the Quest, and no matter what, it would just get wiped.

**Comment by bugbeersandflagons** [comment]
Same issue here, and quests I already created no longer have the rewards I configured previously. This seems to have been an issue since I upgraded to the latest 5e game system update

**Comment by rtschaefer** [comment]
> Same issue here, and quests I already created no longer have the rewards I configured previously. This seems to have been an issue since I upgraded to the latest 5e game system update

I have switched to Forien's Quest module, and it works really well. What a headache to invest time writing story content and details for a quest, only to have the entire content evaporate.  Monk's modules are all brilliant but this is a really unfortunate error.

**Comment by ironmonk88** [comment]
Sorry for the slow response.

The data hasn't been deleted, it just wasn't converted properly.  So you should be able to restore it without issue.
I've added a new function that you can access from the console window to help.

So, load up the quest in question using Enhanced Journal, then open the console window, and run the following command `game.MonksEnhancedJournal.convertReward()`.  That should restore the old data.
Be aware this will overwrite any reward changes you've added since the data disappeared.

If there are any errors in the console or the data doesn't re-appear, let me know and I'll investigate it further.

**Comment by Gazook89** [comment]
This is an older issue (7 months or so) but I feel like I'm bumping into the same issue.  I have created one test quest, which worked fine for whatever reason, but any subsequent quest ends up not saving changes (player permissions, description, active status, etc).  

I get this error:

```
Uncaught (in promise) TypeError: can't convert undefined to object
[Detected 1 package: monks-enhanced-journal]
    _getSubmitData https://assets.forge-vtt.com/bazaar/modules/monks-enhanced-journal/1.0.61/sheets/QuestSheet.js:294
    _onSubmit https://assets.forge-vtt.com/bazaar/modules/monks-enhanced-journal/1.0.61/sheets/EnhancedJournalSheet.js:725
    _onChangeInput https://mirahandel.forge-vtt.com/scripts/foundry.js:3901
    jQuery 2
QuestSheet.js:294:31
```

**Comment by Gazook89** [comment]
Closed PR #309 since it looks like the same fix was applied by @ironmonk88 on 9f38b05cf7f81e3495a9618122fd27128db33d86 just recently.  Not sure if this Issue can be closed as a result.

**Comment by ironmonk88** [comment]
Closing this issue as there was a workaround and shouldn't be an issue right now.
